### ISSUE TITLE
system table for source health status

### ISSUE THREAD

**Issue by neverchanje** [risingwavelabs/risingwave#17163]
### Is your feature request related to a problem? Please describe.

It's very common that the source is connectible but somehow just not consumable.
In this case, we need a interface to see what's going on rather than checking the system logs.
Even as a system admin of RisingWave, I would feel frustrating to check the logs, not to mention the users.	

### Describe the solution you'd like

Either adding informations to the existing table `rw_sources` or introducing a new system table called `rw_source_status` will be fine to me.

- error_message: text
- status: 'healthy' | 'unhealthy'

**Comment by BugenZhao** [comment]
Just to clarify, `source` is not a _physical_ concept until #16003 get completed. It's possible that one source executor encounters failure, while another executor for the same source (in another streaming job) is still healthy.

**Comment by st1page** [comment]
> Just to clarify, `source` is not a _physical_ concept until #16003 get completed. It's possible that one source executor encounters failure, while another executor for the same source (in another streaming job) is still healthy.

Yes, and since shared source needs to implement the backfill operator for each kind of source. We must have some kinds of sources that don't implement shared source for a long time. so the system table's item must be a physical source.

**Comment by st1page** [comment]
RFC https://risingwave-labs.notion.site/RFC-Use-state-table-to-implement-system-table-for-source-health-status-8c006a577052479a8a9409037d19d3c6?pvs=4
